doc/api/diagnostics_channel.md

```diff
@@ -409,6 +459,534 @@ channel.subscribe(onMessage);
+* `thisArg` {any} The receiver to be used for the function call
+* `...args` {any} Optional arguments to pass to the function
+* Returns: {Promise} Chained from promise returned by the given function
+
+Trace a promise-returning function call. This will always produce a
+[`start` event][] and [`end` event][] around the synchronous portion of the
+function execution, and will produce an [`asyncStart` event][] and
+[`asyncEnd` event][] when a promise continuation is reached. It may also
+produce an [`error` event][] if the given function throws an error or the
+returned promise rejects. This will run the given function using
+[`channel.runStores(data, ...)`][] on the `start` channel.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.tracePromise(async () => {
+  // Do something
+}, {
+  some: 'thing',
+});
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.tracePromise(async () => {
+  // Do something
+}, {
+  some: 'thing',
+});
+```
+
+#### `tracingChannel.traceCallback(fn[, position[, data[, thisArg[, ...args]]]])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `fn` {Function} callback using function to wrap a trace around
+* `position` {number} Zero-indexed argument position of expected callback
+* `data` {Object} Shared object to correlate trace events through
+* `thisArg` {any} The receiver to be used for the function call
+* `...args` {any} Optional arguments to pass to the function
+* Returns: {any} The return value of the given function
+
+Trace a callback-receiving function call. This will always produce a
+[`start` event][] and [`end` event][] around the synchronous portion of the
+function execution, and will produce a [`asyncStart` event][] and
+[`asyncEnd` event][] around the callback execution. It may also produce an
+[`error` event][] if the given function throws an error or the returned
+promise rejects. This will run the given function using
+[`channel.runStores(data, ...)`][] on the `start` channel.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.traceCallback((arg1, callback) => {
+  // Do something
+  callback(null, 'result');
+}, 1, {
+  some: 'thing',
+}, thisArg, arg1, callback);
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.traceCallback((arg1, callback) => {
+  // Do something
+  callback(null, 'result');
+}, {
+  some: 'thing',
+}, thisArg, arg1, callback);
+```
+
+### TracingChannel Channels
+
+A TracingChannel is a collection of several channels representing specific
+points in the execution lifecycle of a single traceable action. The behaviour
```

Comment:
`traceable action` and `traceable function` probably need to be defined somewhere. In some places they also _seem_ like they're used interchangeably.

Reply:
The `traceable action` terminology is somewhat _intentionally_ vague to avoid calling it a "span" when there's not really any universally accepted name for what people do with it. 😅